### ISSUE TITLE
swc custom type fails to load

### ISSUE THREAD

**Issue by eleftherioszisis** [BlueBrain/NeuroM#735]
Hi,
I tried to load hundreds of standardized glial cells in `neurom` from `neuromorpho` but the loader fails. Example file: [A1-CA1-L-A63x1zACR3.CNG.swc.zip](https://github.com/BlueBrain/NeuroM/files/2945440/A1-CA1-L-A63x1zACR3.CNG.swc.zip)

I think this is because of the partial support of `neurom` for the swc types (http://www.neuronland.org/NLMorphologyConverter/MorphologyFormats/SWC/Spec.html). Would it be possible to "relax" the `neurom` types, by allowing the user to define a custom type?

``` python
import neurom
nrn = neurom.load_neuron('A1-CA1-L-A63x1zACR3.CNG.swc')
```
``` python
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-2-d35986bfd206> in <module>
----> 1 nrn = neurom.load_neuron('A1-CA1-L-A63x1zACR3.CNG.swc')

~/Projects/ASTROCYTE_SYTHESIS/venv/lib/python3.7/site-packages/neurom/io/utils.py in load_neuron(handle, reader)
    120     else:
    121         name = None
--> 122     return FstNeuron(rdw, name)
    123 
    124 

~/Projects/ASTROCYTE_SYTHESIS/venv/lib/python3.7/site-packages/neurom/fst/_core.py in __init__(self, data_wrapper, name)
     43     def __init__(self, data_wrapper, name='Neuron'):
     44         self._data = data_wrapper
---> 45         neurites, sections = make_neurites(self._data)
     46         soma_check, soma_class = _SOMA_CONFIG[self._data.fmt]
     47         soma = make_soma(self._data.soma_points(), soma_check, soma_class)

~/Projects/ASTROCYTE_SYTHESIS/venv/lib/python3.7/site-packages/neurom/fst/_core.py in make_neurites(rdw)
     87                           points=rdw.data_block[sec.ids],
     88                           section_type=_TREE_TYPES[sec.ntype])
---> 89                   for i, sec in enumerate(rdw.sections))
     90 
     91     # One pass over nodes to connect children to parents

~/Projects/ASTROCYTE_SYTHESIS/venv/lib/python3.7/site-packages/neurom/fst/_core.py in <genexpr>(.0)
     87                           points=rdw.data_block[sec.ids],
     88                           section_type=_TREE_TYPES[sec.ntype])
---> 89                   for i, sec in enumerate(rdw.sections))
     90 
     91     # One pass over nodes to connect children to parents

IndexError: tuple index out of range
```

**Comment by lidakanari** [comment]
This is related to the neurite type (type=7 in file). These neurites should be loaded as undefined by default.

**Comment by eleftherioszisis** [comment]
`undefined` type is 0 and `custom` 7. Should we merge these two as one generic type?

**Comment by eleftherioszisis** [comment]
After discussion with @lidakanari I think the best way to have the neurite types would be:
```
undefined: 0 (Including any type that is not an already a defined type)
soma: 1
axon: 2
basal_dendrite: 3
apical_dendrite: 4
custom: 7
```
5 and 6 are reserved for fork points and terminations and should fall under the undefined neurite type, because they define a point property, not a neurite one.

@wizmer is it possible for such a type spec to be implemented in `MorphIO`?

In addition, is it possible for `.all` to be something else than a positive integer type? For example it could be -1, or something outside the class type numbering.

**Comment by wizmer** [comment]
@eleftherioszisis In your file, is the custom type a dendrite ?

**Comment by eleftherioszisis** [comment]
> @eleftherioszisis In your file, is the custom type a dendrite ?

It is a neurite type yes, i.e. the whole tree is of type 7.

**Comment by wizmer** [comment]
But I mean, it is a dendrite right ? Not an axon ?

**Comment by eleftherioszisis** [comment]
Well, it is an astrocyte, so it is a `glia process` to be more specific.

**Comment by wizmer** [comment]
The standard types in http://www.neuronland.org/NLMorphologyConverter/MorphologyFormats/SWC/Spec.html are

```
Standardized swc files (www.neuromorpho.org) - 
0 - undefined
1 - soma
2 - axon
3 - (basal) dendrite
4 - apical dendrite
5+ - custom
```

After a brief discussion with @eleftherioszisis and @lidakanari, we decided that it would make sense to implement the "custom" type and associate all type id >= 5 with it.
The custom type will be considedered as a 'neurite' (ie. neurom.iter_neurites will iterate over these object) but not as a dendrite.

**Comment by wizmer** [comment]
Implemented in #736
